metrics/influx.go

```diff
@@ -0,0 +1,91 @@
+package metrics
+
+import (
+	"bytes"
+	"time"
+
+	"github.com/containous/traefik/log"
+	"github.com/containous/traefik/safe"
+	"github.com/containous/traefik/types"
+	kitlog "github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/metrics/influx"
+	influxdb "github.com/influxdata/influxdb/client/v2"
+)
+
+var influxClient = influx.New(map[string]string{}, influxdb.BatchPointsConfig{}, kitlog.LoggerFunc(func(keyvals ...interface{}) error {
+	log.Info(keyvals)
+	return nil
+}))
+
+type bufWriter struct {
+	buf    bytes.Buffer
+	config *types.Influx
+}
+
+var influxTicker *time.Ticker
+
+const (
+	influxMetricsReqsName    = "traefik_requests_total"
+	influxMetricsLatencyName = "traefik_request_duration"
+	influxRetriesTotalName   = "traefik_backend_retries_total"
+)
+
+// RegisterInflux registers the metrics pusher if this didn't happen yet and creates a Influx Registry instance.
+func RegisterInflux(config *types.Influx) Registry {
+	if influxTicker == nil {
+		influxTicker = initInfluxTicker(config)
+	}
+
+	return &standardRegistry{
+		enabled:              true,
+		reqsCounter:          influxClient.NewCounter(influxMetricsReqsName),
+		reqDurationHistogram: influxClient.NewHistogram(influxMetricsLatencyName),
+		retriesCounter:       influxClient.NewCounter(influxRetriesTotalName),
+	}
+}
+
+// initInfluxTicker initializes metrics pusher and creates a influxClient if not created already
+func initInfluxTicker(config *types.Influx) *time.Ticker {
+	address := config.Address
+	if len(address) == 0 {
+		address = "localhost:8089"
```

Comment:
https://docs.influxdata.com/influxdb/v0.9/concepts/api/#ports

Reply:
http port default is 8086. We are creating a udp communication here. [https://docs.influxdata.com/influxdb/v1.2/administration/config/#udp](https://docs.influxdata.com/influxdb/v1.2/administration/config/#udp)